bazel/bazel.go

```diff
@@ -23,13 +23,58 @@ import (
 	"io"
 	"os"
 	"os/exec"
+	"path/filepath"
 	"strings"
 
 	blaze_query "github.com/bazelbuild/bazel-watcher/third_party/bazel/master/src/main/protobuf"
 	"github.com/golang/protobuf/proto"
 )
 
-var bazelPath = flag.String("bazel_path", "bazel", "Path to the bazel binary to use for actions")
+var bazelPathFlag = flag.String("bazel_path", "", "Path to the bazel binary to use for actions")
+
+func findBazel() (string) {
+	if len(*bazelPathFlag) > 0 {
+		return *bazelPathFlag
+	} else {
+		path, err := exec.LookPath("bazel")
+		if err == nil {
+			return path
+		}
+
+		// No bazel binary is in the $PATH
+		// When running from the @bazel/ibazel npm package, our binary is
+		// /DIR/node_modules/@bazel/ibazel/bin/$ARCH/ibazel
+		// We can find bazel in
+		// /DIR/node_modules/@bazel/bazel-$ARCH/bazel-0.28.0-linux-x86_64
+		fmt.Printf("didn't find bazel in the $PATH, %v\n", os.Args[0])
+		s := strings.Split(filepath.ToSlash(os.Args[0]), "/")
+		for i := 0; i < len(s); i++ {
+			fmt.Printf("segment %v\n", s[i])
```

Comment:
Can you break the segments down into meaningful names? I have a difficult time reasoning about s[i+2] vs s[i+3]

https://play.golang.org/p/cwPaZv7klcC

Reply:
I'm pretty novice at go so I don't know if there's a way to go straight from a slice like
`a,b,c,d := s[i:i+4]...`

Reply:
I think you can do it this way: https://play.golang.org/p/1f15O3khaB2